### PR TITLE
Add __del__ at Vllm for deleting model

### PR DESCRIPTION
**Pull request by vkehfdl1**
# Description

When using more than one Vllm model in single program, it occurs error. There is a vllm function `destroy_model_parallel()` for preventing this kind of error.

There is a description about this issue at [here](https://github.com/vllm-project/vllm/issues/549#issuecomment-1912976189). 

I tested use two Vllm instance in a single python file at CUDA environment. And it did great. 
Without this code, it occurred error 'data parallel group is already initialized.' 

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- I stared at the code and made sure it makes sense
- Plus, I tested it in my H100 instance with CUDA, and it worked well when I make two `Vllm` instance and run it.

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama_index/llms/vllm.py [hunk 1]
```diff
@@ -186,6 +186,14 @@ def __init__(
             output_parser=output_parser,
         )
 
+    def __del__(self):
+        from vllm.model_executor.parallel_utils.parallel_state import destroy_model_parallel
+        import torch
+        destroy_model_parallel()
```
Comment (hatianzhang) [inline llama_index/llms/vllm.py:192]: will it destroy one of the parallel models?

Comment (vkehfdl1) [reply llama_index/llms/vllm.py:192]: This will destroy `self._client` which is the `VLLM` model.

Comment (vkehfdl1) [reply llama_index/llms/vllm.py:192]: @hatianzhang 

```python
def destroy_model_parallel():
    """Set the groups to none and destroy them."""
    global _TENSOR_MODEL_PARALLEL_GROUP
    if _TENSOR_MODEL_PARALLEL_GROUP:
        torch.distributed.destroy_process_group(_TENSOR_MODEL_PARALLEL_GROUP)
    _TENSOR_MODEL_PARALLEL_GROUP = None
    global _PIPELINE_MODEL_PARALLEL_GROUP
    if _PIPELINE_MODEL_PARALLEL_GROUP:
        torch.distributed.destroy_process_group(_PIPELINE_MODEL_PARALLEL_GROUP)
    _PIPELINE_MODEL_PARALLEL_GROUP = None
    global _PIPELINE_GLOBAL_RANKS
    _PIPELINE_GLOBAL_RANKS = None
```

Here is the code from vllm. 
https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/parallel_utils/parallel_state.py

It looks they save tensor parallel group at global variable when running. 
For using multiple vllm instance, one have to be destroyed.

**Review by hatianzhang — APPROVED** [review]

**Comment by hatianzhang** [comment]
thanks @vkehfdl1 lgtm!

**Comment by logan-markewich** [comment]
@vkehfdl1 apologies, with v0.10.0 the project structure has changed slightly

I can take a stab at porting this PR in a bit, if you don't get to it first :)

**Comment by hatianzhang** [comment]
address refactoring here https://github.com/run-llama/llama_index/pull/10733

**Pull request closed without merging**